### PR TITLE
feat: part-2 refresh session: network UserSessionManager

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

there is no way to get/update user network configration 

### Solutions
in network `UserSessionManager` interface 
```kotlin 
interface UserSessionManager {
    fun userConfig(): Pair<SessionDTO, BackendConfig>
    fun updateSession(accessToken: AccessTokenDTO, refreshTokenDTO: RefreshTokenDTO?): SessionDTO
}
```
that will be implemented by logic

### Dependencies (Optional)
#302 

----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: network/src/commonMain/kotlin/com/wire/kalium/network/session/UserSessionManager.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import io.ktor.client.HttpClientConfig
+import io.ktor.client.plugins.auth.Auth
+import io.ktor.client.plugins.auth.providers.BearerTokens
+import io.ktor.client.plugins.auth.providers.bearer
+
+interface UserSessionManager {
+    fun userConfig(): Pair<SessionDTO, BackendConfig>
+    fun updateSession(accessToken: AccessTokenDTO, refreshTokenDTO: RefreshTokenDTO?): SessionDTO
+    fun onSessionExpiry()
+}
```
Comment (typfel) [inline network/src/commonMain/kotlin/com/wire/kalium/network/session/UserSessionManager.kt:18]: What do you think about dropping the `User` prefix? I think mostly leads to confusing when talking about users in the context of sessions.

```suggestion
interface SessionManager {
    fun session(): Pair<SessionDTO, BackendConfig>
    fun updateSession(accessToken: AccessTokenDTO, refreshTokenDTO: RefreshTokenDTO?): SessionDTO
    fun onSessionExpired()
}
```

**Review by typfel — CHANGES_REQUESTED** [review]

File: network/src/commonMain/kotlin/com/wire/kalium/network/session/UserSessionManager.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
+package com.wire.kalium.network.session
```
Comment (typfel) [inline network/src/commonMain/kotlin/com/wire/kalium/network/session/UserSessionManager.kt:1]: Rename the file as well.

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/user/logout/LogoutApi.kt [hunk 3]
```diff
@@ -10,14 +11,14 @@ import io.ktor.client.request.setBody
 interface LogoutApi {
     suspend fun logout(): NetworkResponse<Unit>
     suspend fun removeCookiesByIds(removeCookiesByIdsRequest: RemoveCookiesByIdsRequest): NetworkResponse<Unit>
-    suspend fun removeCookiesByLabels(removeCookiesWithIdsRequest: RemoveCookiesByLabels): NetworkResponse<Unit>
+    suspend fun removeCookiesByLabels(removeCookiesByLabelsRequest: RemoveCookiesByLabels): NetworkResponse<Unit>
 }
 
-class LogoutImpl(private val httpClient: HttpClient, private val refreshToken: String) : LogoutApi {
+class LogoutImpl(private val httpClient: HttpClient, private val userSessionManager: UserSessionManager) : LogoutApi {
```
Comment (typfel) [inline network/src/commonMain/kotlin/com/wire/kalium/network/api/user/logout/LogoutApi.kt:17]: Not updated after renaming.

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/user/logout/LogoutApi.kt [hunk 4]
```diff
@@ -10,14 +11,14 @@ import io.ktor.client.request.setBody
...
 
     override suspend fun logout(): NetworkResponse<Unit> = wrapKaliumResponse {
         httpClient.post("$PATH_ACCESS/$PATH_LOGOUT") {
-            header(HEADER_KEY_COOKIE, refreshToken)
+            header(HEADER_KEY_COOKIE, userSessionManager.userConfig().first.refreshToken)
```
Comment (typfel) [inline network/src/commonMain/kotlin/com/wire/kalium/network/api/user/logout/LogoutApi.kt:21]: Not updated after renaming.

File: network/src/commonMain/kotlin/com/wire/kalium/network/AuthenticatedNetworkContainer.kt [hunk 5]
```diff
@@ -25,22 +23,18 @@ import com.wire.kalium.network.api.user.logout.LogoutApi
...
+import com.wire.kalium.network.session.installAuth
 import io.ktor.client.engine.HttpClientEngine
-import io.ktor.client.plugins.auth.Auth
-import io.ktor.client.plugins.auth.providers.BearerTokens
-import io.ktor.client.plugins.auth.providers.bearer
 
 class AuthenticatedNetworkContainer(
-    private val backendConfig: BackendConfig,
-    private val sessionDTO: SessionDTO,
+    private val userSessionManager: UserSessionManager,
```
Comment (typfel) [inline network/src/commonMain/kotlin/com/wire/kalium/network/AuthenticatedNetworkContainer.kt:31]: Not updated after renaming.

**Review by typfel — CHANGES_REQUESTED** [review]
Re-naming is not complete

**Comment by MohamadJaara** [comment]
@typfel this PR is not supposed to pass CI (since `SessionManager` is missing in logic) to avoid having 500+ changes PR 
I did run network tests locally, and it worked flawlessly.

**Review by typfel — APPROVED** [review]

**Pull request merged**